### ISSUE TITLE
Error building with zig `0.11.0-dev.2531+aeae71f46`

### ISSUE THREAD

**Issue by charlieroth** [renerocksai/gpt4all.zig#1]
Saw your post on Zig News and thought I would give this a shot! 

I followed the instructions from in `README.md` and ran into the following error and was wondering if you could help me? I have included what I believe to be all of the relevant information to help diagnose the issue. Please let me know if you need any other information 🙂

---

### Machine

Machine: MacBook Pro 2020
Chip: Apple M1
Memory: 8GB
OS Version: `13.3.1 (22E261)`
Xcode Version: `14.3 (14E222b)`

---

### `zig version`

```bash
(base) ❯ zig version
0.11.0-dev.2531+aeae71f46
```

---

### `clang --version`

```bash
(base) ❯ clang --version
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: arm64-apple-darwin22.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

---

### Error From: `zig build -Doptimize=ReleaseFast`

```bash
(base) ❯ zig build -Doptimize=ReleaseFast
zig build-exe chat ReleaseFast native: error: error(compilation): clang failed with stderr: /Users/charlie/github.com/renerocksai/gpt4all.zig/src/llm/ggml.c:9778:13: error: call to undeclared library function 'snprintf' with type 'int (char *, unsigned long, const char *, ...)'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
/Users/charlie/github.com/renerocksai/gpt4all.zig/src/llm/ggml.c:9778:13: note: include the header <stdio.h> or explicitly provide a declaration for 'snprintf'


zig build-exe chat ReleaseFast native: error: the following command failed with 1 compilation errors:
/Users/charlie/github.com/ziglang/zig/zig build-exe /Users/charlie/github.com/renerocksai/gpt4all.zig/src/main.zig -cflags -std=c11 -D_POSIX_C_SOURCE=199309L -pthread -- /Users/charlie/github.com/renerocksai/gpt4all.zig/src/llm/ggml.c -cflags -std=c++11 -- /Users/charlie/github.com/renerocksai/gpt4all.zig/src/llm/utils.cpp /Users/charlie/github.com/renerocksai/gpt4all.zig/src/llm/chat.cpp /Users/charlie/github.com/renerocksai/gpt4all.zig/zig-cache/o/af6ee27f81524e408f1282e171bdf2f4/libcurl.a /Users/charlie/github.com/renerocksai/gpt4all.zig/zig-cache/o/13cfc63e4e965bdf814c77ff25eb93cb/libssh2.a /Users/charlie/github.com/renerocksai/gpt4all.zig/zig-cache/o/08efdba102de56ffd310dbb08c9755d0/libmbedtls.a /Users/charlie/github.com/renerocksai/gpt4all.zig/zig-cache/o/e7a6f07491ff762d359ed77b94201d17/libz.a -lc++ -lc -OReleaseFast --cache-dir /Users/charlie/github.com/renerocksai/gpt4all.zig/zig-cache --global-cache-dir /Users/charlie/.cache/zig --name chat -I /Users/charlie/github.com/renerocksai/gpt4all.zig/src/llm -I /Users/charlie/github.com/renerocksai/gpt4all.zig/src/zig-libcurl/curl/include -D CURL_STATICLIB=1 --enable-cache --listen=-
Build Summary: 4/7 steps succeeded; 1 failed (disable with -fno-summary)
install transitive failure
└─ install chat transitive failure
   └─ zig build-exe chat ReleaseFast native 1 errors
      └─ zig build-lib curl ReleaseFast native success 1s MaxRSS:78M
         ├─ zig build-lib ssh2 ReleaseFast native success 578ms MaxRSS:77M
         │  └─ zig build-lib mbedtls ReleaseFast native success 1s MaxRSS:79M
         ├─ zig build-lib mbedtls ReleaseFast native (reused)
         └─ zig build-lib z ReleaseFast native success 1s MaxRSS:73M
/Users/charlie/github.com/renerocksai/gpt4all.zig/src/llm/ggml.c:1:1: error: unable to build C object: clang exited with code 1
```

**Comment by charlieroth** [comment]
I was able to build and run the program by changing line `34` in `build.zig` to a more recent version of `-D_POSIX_C_SOURCE`. I found this version value at this link https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html

I don't know much about C/C++ development or I would submit a PR with a proper explanation but I hope this helps understanding my issue

Git Diff
```
     // now, here comes the C and C++ stuff for the actual chat client
     exe.addIncludePath("src/llm");
-    exe.addCSourceFile("src/llm/ggml.c", &.{ "-std=c11", "-D_POSIX_C_SOURCE=199309L", "-pthread" });
+    exe.addCSourceFile("src/llm/ggml.c", &.{ "-std=c11", "-D_POSIX_C_SOURCE=200809L", "-pthread" });
     exe.addCSourceFiles(&.{ "src/llm/utils.cpp", "src/llm/chat.cpp" }, &.{"-std=c++11"});
     curl.link(exe, .{});
     exe.linkLibC();
```

**Comment by renerocksai** [comment]
Thank you! After some digging, I noticed the vendored-in mbedtls code also uses this value. When I encountered the error, I found my Linux answer [in the gettime() man page](https://man7.org/linux/man-pages/man2/clock_gettime.2.html), which apparently doesn't hold true for all versions of macOS.
Interestingly enough, I looked into the Makefile of the original C++ project, no `-D_POSIX_C_SOURCE` is mentioned there.

Fixed now. Thanks a lot for pointing this out and finding the solution!!! :+1:
